### PR TITLE
cmd: add dial timeout and verify sink connection status

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add dial timeout and verify sink connection when creating changefeed.

Cc https://github.com/pingcap/ticdc/issues/542

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Add client dial timeout
- Verify sink connection when creating changefeed

### CODE REVIEW

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -297,6 +297,11 @@ func newMySQLSink(ctx context.Context, sinkURI *url.URL, dsn *dmysql.Config, fil
 	if err != nil {
 		return nil, errors.Annotatef(err, "Open database connection failed, dsn: %s", dsnStr)
 	}
+	err = db.PingContext(ctx)
+	if err != nil {
+		return nil, errors.Annotatef(err, "fail to open MySQL connection")
```
[the remark this answers is not in the document]
Comment (overvenus) [reply cdc/sink/mysql.go:302]: It could leak the password.

[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/sink/mysql.go:302]: Oh, yes. Seems `dsnStr` in L298 also needs to be removed

**Pull request merged**